internal/webhook/podmutation/webhookhandler_test.go

```diff
@@ -123,6 +123,7 @@ func TestShouldInjectSidecar(t *testing.T) {
 			},
 		},
 	} {
+		tt := tt
```

Comment:
why is this necessary?

Reply:
I didn't detect the previous lint error because of the golang-ci version. I upgraded to the latest and now I'm getting a lot of `Implicit memory aliasing in for loop` errors.
I fixed them because we will have to do that in the future anyway.

Reply:
I thought that would only happen if we were using a reference to the test's memory address? Is this a known go issue for the testing library?

Reply:
In general, we do need this assignment. It doesn't matter unless we enable `t.Parallel()` for the test, but it's a good practice and probably why linters are complaining about it. Fundamentally, we're using the `tt` variable in a closure via `t.Run`, so if we ran the cases in parallel without reassigning, we'd get the same value in all of them. See https://go.dev/blog/loopvar-preview for a clearer and more detailed explanation.

Reply:
Once we upgrade the minimum Go version to 1.22 in our go.mod, yes. Just compiling with Go 1.22 isn't enough according to the blog post.